### PR TITLE
Use unix socket in testserver

### PR DESCRIPTION
**Pull request by mberhault**
Remove all the stdout parsing stuff.
PGURL waits for the socket to exist and loops forever until it does.

### CODE REVIEW

File: testserver/testserver.go [hunk 1]
```diff
@@ -108,7 +110,9 @@ func NewTestServer() (*TestServer, error) {
 		cockroachBinary = "cockroach"
 	}
 
-	baseDir, err := ioutil.TempDir("", "cockroach-testserver")
+	// Force "/tmp/" so avoid OSX' really long temp directory names
```
Comment (spencerkimball) [inline testserver/testserver.go]: s/OSX'/OSX's/

File: testserver/testserver.go [hunk 2]
```diff
@@ -118,6 +122,17 @@ func NewTestServer() (*TestServer, error) {
 		return nil, fmt.Errorf("could not create logs directory: %s: %s", logDir, err)
 	}
 
+	options := url.Values{
+		"host": []string{baseDir},
```
Comment (spencerkimball) [inline testserver/testserver.go]: Surprised the pgurl works like this with sockets.

File: testserver/testserver.go (hunk 1 above)
Comment (mberhault) [inline testserver/testserver.go]: Done

**Comment by spencerkimball** [comment]
LGTM

File: testserver/testserver.go (hunk 2 above)
Comment (mberhault) [inline testserver/testserver.go]: yeah, it's really gross. That's the reason I've kept the `--socket` flag hidden for now. If there's real demand for it, we can make it more publicly-visible and more in line with postgres, but for now, this is much simpler to use, especially for the dynamic case.

**Pull request merged**